### DISCUSSION TITLE
Creating a standalone plugin with Generator

### DISCUSSION THREAD

**Discussion by joshuapease** [craftcms/generator#33] [Ideas] [answered]
More of a question than an Idea, but is there a way to use Generator without needing to install & scaffold and install a Craft CMS site? 

I'm starting a new standalone plugin that will be published to the Craft Plugin Store. 

I can't figure out how to run `craft make plugin` without spinning up a new Craft site in my plugin repo.

**Comment by AugustMiller** [comment, marked as answer]
Currently, no. Generator is intended to be used inside a Craft application, as a good deal of its functionality is aware of the installation itself—just like a plugin! If you take a look at some of the generators, you’ll notice that it dynamically prompts for, discovers, validates, and includes a number of component names and classes. This tight coupling means Generator's output can be extremely accurate and consistent… and provide immediate feedback when something is misconfigured.

Plugins and modules ought to be designed and developed inside a Craft application… and even if they could be scaffolded independently, there is very little progress that can be made without propping up a development environment.

The [Plugin Factory project](https://github.com/nystudio107/generator-craftplugin) from @nystudio107 was based on [Yeoman](http://yeoman.io) so that it could be run without a back-end.

**Reply by joshuapease** [reply]
Thanks for that feedback. I've been using your recommended setup and it's working well for me.

Just to make sure I'm understanding 💯, you recommending something like this.

```
- Craft Project (not tracked in git)
  ∟ plugins (sub-directory)
    ∟plugin-a (git repo)
    ∟plugin-b (git repo)
```

**Reply by brandonkelly** [reply]
@joshuapease That setup works if these are private plugins that are only being built for this one project.

If you plan to publish them to the Plugin Store, or share them with other projects, you’ll want to have them live _outside_ of the Craft project, where they can be made into their own Git repos.

**Reply by joshuapease** [reply]
Would it work for public projects if the Craft project isn't tracked by git but each plugin folder is it's own public git repo?

That's what I've been doing locally and so far it's worked exactly as I'd hoped.

**Reply by brandonkelly** [reply]
Yep, that would be fine!

**Reply by joshuapease** [reply]
👍  I love `generator` so much
